enhancements/network/user-defined-network-segmentation.md

```diff
@@ -379,11 +382,283 @@ any user-defined primary network.
+|-------------------|---------------------------------------------------------------------------------------------------------------|----------|
+| NamespaceSelector | List of the standard `metav1.LabelSelector` selector for which namespace the network should be available for. | No       |
+| Template          | The user defined network spec.                                                                                | No       |
+
+The template type should be the namespace scope CRD spec.
+
+> **Note:** The spec should be extended with care and strive to have minimal set of fields to provide nice abstraction for the NAD spec.
+
+##### Status
+The CRD status should reflect the NAD state through conditions.
+For example, when the NAD its created the condition should be placed with status `True`.
+
+For cluster scoped networks, the condition should be true once all desired namespaces are provisioned with the corresponding NAD.
+
+The cluster scoped CRD status should reflect on which namespaces the network is available.
+
+##### CRD Definitions
+
+The following CRDs should be installed by [CNO](https://github.com/openshift/cluster-network-operator) and managed by OVN-Kubernetes.
+
+##### Namespace scoped CRD
+```golang
+// UserDefinedNetwork describe network request for a Namespace
+type UserDefinedNetwork struct {
+    metav1.TypeMeta   `json:",inline"`
+    metav1.ObjectMeta `json:"metadata,omitempty"`
+    // +kubebuilder:validation:Required
+    // +kubebuilder:validation:XValidation:rule="self == oldSelf", message="Spec is immutable"
+    // +kubebuilder:validation:XValidation:rule=" self.topology != 'Layer3' || (self.topology == 'Layer3' && size(self.subnets) > 0)", message="Subnets is required for Layer3 topology"
+    // +kubebuilder:validation:XValidation:rule="self.topology != 'Localnet' || (self.topology == 'Localnet' && self.role == 'Secondary')", message="Topology localnet is not supported for primary network"
+    // +kubebuilder:validation:XValidation:rule="!has(self.ipamLifecycle) || (self.ipamLifecycle == 'Persistent' && (self.topology == 'Layer2' || self.topology == 'Localnet'))", message="ipamLifecycle is supported for Layer2 and Localnet topologies"
+    // +required
+    Spec UserDefinedNetworkSpec `json:"spec"`
+    // +optional
+    Status UserDefinedNetworkStatus `json:"status,omitempty"`
+}
+```
+Suggested API validation rules:
+- Spec defined as immutable 
+  <br>
+  Avoid incomplete state in a scenario where a NAD is created according to UDN spec, and pods connected to the network, 
+  UDN spec changes, now pods connected to a network that was created from previous revision spec.
+- `Subnets` are mandatory for `Layer3` topology.
+- `Localnet` topology is not supported for primary network.
+- `IPAMLifecycle` is supported for `Layer2` and `Localnet` topology.
+
+Suggested CRD short-name: `udn`
```

Comment:
How many of these will exist in a cluster? How many users do you expect to interact with these? We typically try to reserve short names for very commonly used resources, since they create conflicts.

Sig CLI officially only support fully qualified resource names for this reason, and expect short names to be only a lightly used feature for the most frequently interacted with resources

Reply:
I think we can just use the full name